### ISSUE TITLE
Allow empty/null type path ?

### ISSUE THREAD

**Issue by vinilios** [emberjs/data#2844]
The current default behaviour of rest adapter buildURL method appends the result of pathForType for the given type and joins it along with other params in order to construct the final resource request url. In case pathForType returns an empty or null value (this could be a case for an API which serve a resource results from the root url) the resulted url will contain an additional slash causing issues if such urls are not handled server-side.

```javascript
[host, typePath, id].join("/") // result is host.com/api/v1//id for null/empty typePath
```

Is this a case that you would consider ember-data should deal with ? The change is trivial, i could provide a PR including the corresponding tests, but i'm not sure what kind of side-effects and/or backwards compatible issues such a change might cause.

**Comment by bmac** [comment]
@vinilios if you'd like to submit a pr for this feature then go for it.

**Comment by vinilios** [comment]
do i have to branch ?

**Comment by bmac** [comment]
Closed by #2946.
